docs/UserGuide.md

```diff
@@ -159,21 +159,36 @@ Examples:
 
 ### Locating persons by name: `find`
 
-Finds persons whose names contain any of the given keywords.
-
-Format: `find KEYWORD [MORE_KEYWORDS]`
-
-* The search is case-insensitive. e.g `hans` will match `Hans`
-* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
-* Only the name is searched.
-* Only full words will be matched e.g. `Han` will not match `Hans`
-* Persons matching at least one keyword will be returned (i.e. `OR` search).
-  e.g. `Hans Bo` will return `Hans Gruber`, `Bo Yang`
+Finds the employees that satisfy the find requirements based on attributes.
+
+Format: `find [i/KEYWORD] [n/KEYWORD [OTHER KEYWORD]] [p/KEYWORD] [e/KEYWORD [OTHER KEYWORD]] [d/KEYWORD [OTHER KEYWORD]] [r/KEYWORD [OTHER KEYWORD]] [s/LOWERBOUND - UPPERBOUND]`
+
+* Find command must be used with at least one prefix and the keyword for the prefix must not be empty.
+* Refer to the prefix summary to understand which employee attribute corresponds with which prefix.
+* Find command for all attributes are case-insensitive. 
+* For finding by ID and Phone number, the command will find employees whose ID and Phone number contains the respective keywords.
+* For finding by Name, Email, Department and Role, the order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
+* For finding by Name, Department and Role, only full words will be matched e.g. `Han` will not match `Hans` and `Fina` will not match `Finance`.
+* For finding by Email, Email will be matched if it contains the letter sequence e.g. `alex` will match `alexyeoh@example.com`.
+* For finding by Name, Email, Department and Role, matching at least one keyword will be returned (i.e. `OR` search).
+  e.g. `Hans Bo` will return `Hans Gruber`, `Bo Yang` e.g. `alex yeoh` will return `alextan@example.com`, `berniceyeoh@example.com`.
+* For finding by Salary, command will find employees whose salaries fall within the range of LOWERBOUND to UPPERBOUND inclusive of LOEWRBOUND and UPPERBOUND.
```

Comment:
The find command appears to not parse decimal values e.g. `find s/5000.00 - 5100.00`. Perhaps it will be good to mention in the UG that the salary field can only be searched with integer values, or the code modified to accept floating values?

Reply:
Ok yes I mentioned in the UG that lower bound and upper bound should be a non negative integer